extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/sys/health/VaultHealthStatus.java

```diff
@@ -0,0 +1,100 @@
+package io.quarkus.vault.runtime.sys.health;
```

Comment:
this is part of the api. it should be in package `io.quarkus.vault.sys`

Reply:
I was thinking the same but, there are a lot of sys operations so I decided to split but we can put together for sure.

Reply:
not sure I got you. everything that is either a param or a return type in the system backend engine interface needs to be in a non runtime package. that's the rule I got from Guillaume when I initially pushed the extension. this applies to:
 - VaultInit
 - VaultHealth
 - VaultHealthStatus 
 - VaultSealStatus
also I saw that you decided to not allow passing params to `VaultHealth health();` whereas the vault api allows to pass a bunch of args: standbyok, perfstandbyok, activecode, standbycode, drsecondarycode, performancestandbycode, sealedcode 
is that on purpose?

Reply:
Yes it is on purpose. As for me, it sounds strange that someone might want to change the status code definition and the status flags can be configured in `application.properties`. I was thinking more about simplicity/common cases and if someone requests it we can always extend it easily.

Reply:
> As for me, it sounds strange that someone might want to change the status code definition

some time ago I had to integrate vault into k8s. the situation was different because I was implementing the liveness/readiness probes of vault itself (not a client of vault). but still it shows a situation where I had to change the status codes. basically I turned everything to 200 for liveness, otherwise vault would have been killed and restarted anytime its state would have bee something other than the ok state. which would have been counter productive because no restart would have helped. and for the readiness I did the same except for the sealedcode because when vault is sealed I did not want it to be part of the service. but I still wanted it to be reachable when it is not initialized. because if it is, all vaults are in the same situation, and making it not ready will just make it harder to access it.
see [the probe definitions.](https://github.com/vsevel/vault/blob/master/demoit/4-vault/vault-deployment.yaml#L64-L69)
you have also to take into account that people sometime use the same endpoint for liveness and readiness, which is a very bad idea when integrating external systems into the app health checks (this could lead to unwanted app restarts, and cascading crash loop backoffs).

are you expecting your healthchecks to be used for readiness only?

Reply:
I understand but as you said this is different, we are not integrating Vault directly but the app which we are already taking care of the status code (200) so one might not need to tune this parameter.

By default yes, if people wants to use it as liveness they can implement by themselves using the programmatic way.